src/test/java/com/xceptance/neodymium/tests/NeodymiumTest.java

```diff
@@ -52,29 +54,106 @@ public static void deleteTempFile(File tempFile)
+            {
+                String methodName = result.getFailures().get(i).getDescription().getMethodName();
+                Assert.assertEquals("Failure message", expectedFailureMessages.get(methodName), result.getFailures().get(i).getMessage());
+            }
+        }
+        catch (AssertionError e)
         {
-            Assert.assertTrue("Failure count", expectedFailCount == 1);
-            Assert.assertEquals("Failure message", expectedFailureMessage, result.getFailures().get(0).getMessage());
+            throw new RuntimeException("Assertion failed. " + e.getMessage() + " Stack trace: " + stackTrace, e);
         }
     }
 
-    public void checkPass(Result result, int expectedRunCount, int expectedIgnoreCount, int expectedFailCount)
+    /**
+     * Assert all tests in run were successful
+     * 
+     * @param result
+     *            test result to make assertions on
+     * @param expectedRunCount
+     *            expected number of run tests (including ignored)
+     * @param expectedIgnoreCount
+     *            expected number of ignored tests
+     */
+    public void checkPass(Result result, int expectedRunCount, int expectedIgnoreCount)
     {
-        check(result, true, expectedRunCount, expectedIgnoreCount, expectedFailCount, null);
+        check(result, true, expectedRunCount, expectedIgnoreCount, 0, null);
     }
 
+    /**
+     * Assert the are one or more failed tests in run
+     * 
+     * @param result
+     *            test result to make assertions on
+     * @param expectedRunCount
+     *            expected number of run tests (including ignored)
+     * @param expectedIgnoreCount
+     *            expected number of ignored tests
+     * @param expectedFailCount
+     *            expected number of failed tests
+     */
+    public void checkFail(Result result, int expectedRunCount, int expectedIgnoreCount, int expectedFailCount)
+    {
+        check(result, false, expectedRunCount, expectedIgnoreCount, expectedFailCount, null);
+    }
+
+    /**
+     * Assert the are one or more failed tests in run
+     * 
+     * @param result
+     *            test result to make assertions on
+     * @param expectedRunCount
+     *            expected number of run tests (including ignored)
+     * @param expectedIgnoreCount
+     *            expected number of ignored tests
+     * @param expectedFailCount
+     *            expected number of failed tests
+     * @param expectedFailureMessage
+     *            expected failure message for all failures
+     */
     public void checkFail(Result result, int expectedRunCount, int expectedIgnoreCount, int expectedFailCount,
                           String expectedFailureMessage)
     {
-        check(result, false, expectedRunCount, expectedIgnoreCount, expectedFailCount, expectedFailureMessage);
+        HashMap<String, String> expectedFailureMessages = new HashMap<String, String>();
+        for (Failure failure : result.getFailures())
```

Comment:
All local variables should be declared `final` unless mutability is explicitly required.